lib/machine-loader.js

```diff
@@ -87,9 +87,54 @@ function getMachineName (machineId) {
     .then(it => it.name)
 }
 
-function getMachine (machineId) {
```

Comment:
I think you can still refactor this function, there's only a couple of lines different from the machine names one.